File: ST12_STEPS_DHS_DualSurvey_NCD/03_scripts/run_st12_workflow.py
#!/usr/bin/env python3
"""Orchestrate ST12 analysis: STEPS → DHS → dual synthesis → figures."""
from __future__ import annotations

import runpy
import sys
from pathlib import Path

HERE = Path(__file__).resolve().parent


def run(script: str) -> None:
    path = HERE / script
    print(f"\n{'='*60}\nRunning {script}\n{'='*60}", flush=True)
    runpy.run_path(str(path), run_name="__main__")


def main():
    for s in [
        "01_steps_construct_analyze.py",
        "02_dhs_construct_analyze.py",
        "03_dual_survey_synthesis.py",
        "04_figures.py",
    ]:
        run(s)
    print("\nST12 workflow complete.")


if __name__ == "__main__":
    main()

File: ST12_STEPS_DHS_DualSurvey_NCD/03_scripts/run_st12_workflow.R
#!/usr/bin/env Rscript
# ST12 full R workflow: STEPS → DHS → dual synthesis
# Requires packages in R_LIBS_USER (survey, haven, dplyr, ...)

Sys.setenv(ST12_ROOT = "/mnt/c/Users/HFD 2/Research/02_Studies/ST12_STEPS_DHS_DualSurvey_NCD")
root <- Sys.getenv("ST12_ROOT")
scripts <- file.path(root, "03_scripts")

cat("ST12 R workflow\n")
cat("R version:", as.character(getRversion()), "\n")
cat("R_LIBS_USER:", Sys.getenv("R_LIBS_USER", unset = path.expand("~/R/library")), "\n\n")

source(file.path(scripts, "01_steps_construct_analyze.R"))
source(file.path(scripts, "02_dhs_construct_analyze.R"))
source(file.path(scripts, "03_dual_survey_synthesis.R"))

cat("\n=== ST12 R workflow complete ===\n")
cat("Tables written with _R suffix in 04_tables/\n")
